### PR TITLE
fix: Fix of BOHB and SyncBOHB

### PR DESCRIPTION
**Pull request by mseeger**
Conditions in `KernelDensityEstimator` for whether KDE models can be fit are typically violated in `MultiFidelityKernelDensityEstimator`, so this did the same as ASHA. Here, I am changing the choice of `r_acq` to depend on the "can fit" decision

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mseeger** [comment]
I reran my experiments on our 13 surrogate benchmarks. BOHB is now quite different from ASHA, and often better.

File: syne_tune/optimizer/schedulers/searchers/kde/kde_searcher.py [hunk 1]
```diff
@@ -353,30 +354,27 @@ def acquisition_function(x):
 
         return suggestion
 
-    def _train_kde(self, train_data, train_targets):
-        if train_data.shape[0] < self.num_min_data_points:
+    def _good_data_size(self, data_shape: Tuple[int, int]) -> Optional[int]:
+        num_data, num_features = data_shape
+        n_good = max(self.num_min_data_points, (self.top_n_percent * num_data) // 100)
+        if min(n_good, num_data - n_good) <= num_features:
```
Comment (aaronkl) [inline syne_tune/optimizer/schedulers/searchers/kde/kde_searcher.py:360]: I am not sure I follow here. I assume num_data - n_godd are the number of data point for the KDE that models the poorly performing configuration? Why do both have to be larger or equal than num_features? Also you might want to change the name of the function to reflect that you check the correct sizes for both models?

File: syne_tune/optimizer/schedulers/searchers/kde/kde_searcher.py [hunk 2]
```diff
@@ -353,30 +354,27 @@ def acquisition_function(x):
...
         idx = np.argsort(train_targets)
-
         train_data_good = train_data[idx[:n_good]]
-        train_data_bad = train_data[idx[n_good : n_good + n_bad]]
-
-        if train_data_good.shape[0] <= train_data_good.shape[1]:
-            return None
-        if train_data_bad.shape[0] <= train_data_bad.shape[1]:
-            return None
+        train_data_bad = train_data[idx[n_good:]]
```
Comment (aaronkl) [inline syne_tune/optimizer/schedulers/searchers/kde/kde_searcher.py:375]: Why do we not check anymore if n_bad is greater than self.num_min_data_points?

File: syne_tune/optimizer/schedulers/searchers/kde/kde_searcher.py (hunk 1 above)
Comment (mseeger) [reply syne_tune/optimizer/schedulers/searchers/kde/kde_searcher.py:360]: The current code is doing these checks, I did not change that

File: syne_tune/optimizer/schedulers/searchers/kde/kde_searcher.py (hunk 2 above)
Comment (mseeger) [reply syne_tune/optimizer/schedulers/searchers/kde/kde_searcher.py:375]: This is done in the `_good_data_size` method

**Comment by wesk** [comment]
I just triggered a re-run of the tests, let's see if this is transient or not

**Comment by mseeger** [comment]
Thanks, it was a hickup

**Review by aaronkl — DISMISSED** [review]

**Review by aaronkl — APPROVED** [review]

**Pull request merged**